src/evo/deterministicmns.cpp

```diff
@@ -790,7 +843,7 @@ bool CDeterministicMNManager::BuildNewListFromBlock(const CBlock& block, const C
         } else if (tx.nType == TRANSACTION_QUORUM_COMMITMENT) {
             llmq::CFinalCommitmentTxPayload qc;
             if (!GetTxPayload(tx, qc)) {
-                assert(false); // this should have been handled already
+                return _state.DoS(100, false, REJECT_INVALID, "bad-protx-payload");
```

Comment:
Should this be?
```suggestion
                return _state.DoS(100, false, REJECT_INVALID, "bad-qctx-payload");
```

Reply:
Good catch. I checked the other occurrences of and ended up with this one 997f2084ff67ce7dae4e08e886f90889139dfbce which fixes your suggestion and also this one from this PR

https://github.com/dashpay/dash/blob/4bfad3acfd8970d17070316eda18a48770a3d3ed/src/evo/cbtx.cpp#L198-L202

and this one which is not coming from this PR but its still wrong

https://github.com/dashpay/dash/blob/4bfad3acfd8970d17070316eda18a48770a3d3ed/src/llmq/quorums_blockprocessor.cpp#L327-L332

There was already a `bad-qc-payload` used in thats why i picked this naming https://github.com/dashpay/dash/blob/4bfad3acfd8970d17070316eda18a48770a3d3ed/src/llmq/quorums_commitment.cpp#L136-L139

Reply:
Yeah, this belongs to #3504 to actually. Cherry-picked there, thanks! 👍